File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/repository/jpa/CakeHeartJpaRepository.java
package com.cakk.domain.mysql.repository.jpa;

import java.util.List;
import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.cakk.domain.mysql.entity.cake.Cake;
import com.cakk.domain.mysql.entity.cake.CakeHeart;
import com.cakk.domain.mysql.entity.user.User;

public interface CakeHeartJpaRepository extends JpaRepository<CakeHeart, Long> {

	List<CakeHeart> findAllByUser(User user);

	Optional<CakeHeart> findByUserAndCake(User user, Cake cake);
}

File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/repository/jpa/CakeHeartJpaRepository.kt
package com.cakk.domain.mysql.repository.jpa

import com.cakk.domain.mysql.entity.cake.Cake
import com.cakk.domain.mysql.entity.cake.CakeHeart
import com.cakk.domain.mysql.entity.user.User
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface CakeHeartJpaRepository : JpaRepository<CakeHeart, Long> {

	fun findAllByUser(user: User): List<CakeHeart>
    fun findByUserAndCake(user: User, cake: Cake): CakeHeart?
    fun existsByUserAndCake(user: User, cake: Cake): Boolean
}

